pkg/query/storeset_test.go

```diff
@@ -130,6 +131,93 @@ func (s *testStores) CloseOne(addr string) {
 	delete(s.srvs, addr)
 }
 
+type mockedInfo struct {
+	info infopb.InfoResp
+}
+
+func (s *mockedInfo) Info(ctx context.Context, r *infopb.InfoReq) (*infopb.InfoResp, error) {
+	return &s.info, nil
+}
+
+type testInfoMeta struct {
+	extlsetFn func(add string) []labelpb.ZLabelSet
+	storeType component.StoreAPI
+	store     infopb.StoreInfo
+	rule      infopb.RulesInfo
+	metadata  infopb.MetricMetadataInfo
+	target    infopb.TargetsInfo
+	exemplar  infopb.ExemplarsInfo
+}
+
+type testInfoSrvs struct {
+	srvs       map[string]*grpc.Server
+	orderAddrs []string
+}
+
+func startInfoSrvs(infoMetas []testInfoMeta) (*testInfoSrvs, error) {
+	info := &testInfoSrvs{
+		srvs: map[string]*grpc.Server{},
+	}
+
+	for _, meta := range infoMetas {
```

Comment:
Need to be careful here and add:
```
meta := meta
```

Otherwise, you'll have problems: https://play.golang.org/p/dCePpqfNcEh

Reply:
😲 Wasn't aware of this behavior! Thank you for pointing out :)